### PR TITLE
product-interface-dynamic-links-mapping

### PR DESCRIPTION
**Pull request by RakeshJesadiya**
Mapping Dynamic links to the all the product interface, Customizable product interface and physical product interface.

## Purpose of this pull request

This pull request (PR) update the bulk of product related pages to update dynamic links on the all types of pages where static interface declared for the product, Customizable product and physical product interface.

## Affected DevDocs pages

https://devdocs.magento.com/guides/v2.3/graphql/interfaces/bundle-product.html
https://devdocs.magento.com/guides/v2.4/graphql/interfaces/bundle-product.html

### CODE REVIEW

**Review by aleron75 — APPROVED** [review]

File: src/guides/v2.3/graphql/interfaces/downloadable-product.md [hunk 1]
```diff
@@ -14,12 +14,12 @@ The `DownloadableProduct` object contains the following attributes:
...
 --- | --- | ---
-`downloadable_product_links` | [`DownloadableProductLinks`] | An array containing information about the links for this downloadable product
-`downloadable_product_samples` | [`DownloadableProductSamples`] | An array containing information about samples of this downloadable product
+`downloadable_product_links` | [[DownloadableProductLinks]](#downloadableProductLinks) | An array containing information about the links for this downloadable product
+`downloadable_product_samples` | [[DownloadableProductSamples]](#downloadableProductSamples)  | An array containing information about samples of this downloadable product
 `links_purchased_separately` | Int | A value of 1 indicates that each link in the array must be purchased separately
 `links_title` | String | The heading above the list of downloadable products
 
-### DownloadableProductSamples object
+### DownloadableProductSamples object {#downloadableProductSamples}
```
Comment (keharper) [inline src/guides/v2.3/graphql/interfaces/downloadable-product.md:22]: Capitalize the "D" and update all references to the anchor.

File: src/guides/v2.3/graphql/interfaces/downloadable-product.md [hunk 2]
```diff
@@ -32,7 +32,7 @@ Attribute | Type | Description
 `sort_order` | Int | A number indicating the sort order
 `title` | String | The display name of the sample
 
-### DownloadableProductLinks object
+### DownloadableProductLinks object {#downloadableProductLinks}
```
Comment (keharper) [inline src/guides/v2.3/graphql/interfaces/downloadable-product.md:35]: Same here.

File: src/guides/v2.3/graphql/interfaces/product-interface-implementations.md [hunk 3]
```diff
@@ -16,6 +16,7 @@ Product type | Implements | Has product-specific attributes?
...
 [SimpleProduct]({{ page.baseurl }}/graphql/interfaces/simple-product.html) | [ProductInterface]({{ page.baseurl }}/graphql/interfaces/product-interface.html), [PhysicalProductInterface]({{ page.baseurl }}/graphql/interfaces/product-interface.html#PhysicalProductInterface), [CustomizableProductInterface]({{ page.baseurl }}/graphql/interfaces/customizable-option-interface.html) | No
 [VirtualProduct]({{ page.baseurl }}/graphql/interfaces/virtual-product.html) | [ProductInterface]({{ page.baseurl }}/graphql/interfaces/product-interface.html),  [CustomizableProductInterface]({{ page.baseurl }}/graphql/interfaces/customizable-option-interface.html) | No
+[GiftCardProduct]({{ page.baseurl }}/graphql/interfaces/gift-card-product.html) | [ProductInterface]({{ page.baseurl }}/graphql/interfaces/product-interface.html),  [PhysicalProductInterface]({{ page.baseurl }}/graphql/interfaces/product-interface.html#PhysicalProductInterface),[CustomizableProductInterface]({{ page.baseurl }}/graphql/interfaces/customizable-option-interface.html) | Yes
```
Comment (keharper) [inline src/guides/v2.3/graphql/interfaces/product-interface-implementations.md:19]: Alphabetize this row

File: src/guides/v2.3/graphql/interfaces/product-interface.md [hunk 4]
```diff
@@ -17,6 +17,7 @@ The `items` that are returned in a `ProductInterface` array can also contain att
    -  [DownloadableProduct]({{ page.baseurl }}/graphql/interfaces/downloadable-product.html)
    -  [GroupedProduct]({{ page.baseurl }}/graphql/interfaces/grouped-product.html)
    -  [VirtualProduct]({{ page.baseurl }}/graphql/interfaces/virtual-product.html)
+   -  [GiftCardProduct]({{ page.baseurl }}/graphql/interfaces/gift-card-product.html)
```
Comment (keharper) [inline src/guides/v2.3/graphql/interfaces/product-interface.md:20]: Alphabetize

File: src/guides/v2.4/graphql/interfaces/downloadable-product.md [hunk 5]
```diff
@@ -13,12 +13,12 @@ The `DownloadableProduct` object contains the following attributes:
...
 --- | --- | ---
-`downloadable_product_links` | [`DownloadableProductLinks`] | An array containing information about the links for this downloadable product
-`downloadable_product_samples` | [`DownloadableProductSamples`] | An array containing information about samples of this downloadable product
+`downloadable_product_links` | [[DownloadableProductLinks]](#downloadableProductLinks) | An array containing information about the links for this downloadable product
+`downloadable_product_samples` | [[DownloadableProductSamples]](#downloadableProductSamples)  | An array containing information about samples of this downloadable product
 `links_purchased_separately` | Int | A value of 1 indicates that each link in the array must be purchased separately
 `links_title` | String | The heading above the list of downloadable products
 
-### DownloadableProductSamples object
+### DownloadableProductSamples object {#downloadableProductSamples}
```
Comment (keharper) [inline src/guides/v2.4/graphql/interfaces/downloadable-product.md:21]: Capitalize the D

File: src/guides/v2.4/graphql/interfaces/downloadable-product.md [hunk 6]
```diff
@@ -31,7 +31,7 @@ Attribute | Type | Description
 `sort_order` | Int | A number indicating the sort order
 `title` | String | The display name of the sample
 
-### DownloadableProductLinks object
+### DownloadableProductLinks object {#downloadableProductLinks}
```
Comment (keharper) [inline src/guides/v2.4/graphql/interfaces/downloadable-product.md:34]: Capitalize

File: src/guides/v2.4/graphql/interfaces/product-interface-implementations.md [hunk 7]
```diff
@@ -15,6 +15,7 @@ Product type | Implements | Has product-specific attributes?
...
 [SimpleProduct]({{ page.baseurl }}/graphql/interfaces/simple-product.html) | [ProductInterface]({{ page.baseurl }}/graphql/interfaces/product-interface.html), [PhysicalProductInterface]({{ page.baseurl }}/graphql/interfaces/product-interface.html#PhysicalProductInterface), [CustomizableProductInterface]({{ page.baseurl }}/graphql/interfaces/customizable-option-interface.html) | No
 [VirtualProduct]({{ page.baseurl }}/graphql/interfaces/virtual-product.html) | [ProductInterface]({{ page.baseurl }}/graphql/interfaces/product-interface.html),  [CustomizableProductInterface]({{ page.baseurl }}/graphql/interfaces/customizable-option-interface.html) | No
+[GiftCardProduct]({{ page.baseurl }}/graphql/interfaces/gift-card-product.html) | [ProductInterface]({{ page.baseurl }}/graphql/interfaces/product-interface.html),  [PhysicalProductInterface]({{ page.baseurl }}/graphql/interfaces/product-interface.html#PhysicalProductInterface),[CustomizableProductInterface]({{ page.baseurl }}/graphql/interfaces/customizable-option-interface.html) | Yes
```
Comment (keharper) [inline src/guides/v2.4/graphql/interfaces/product-interface-implementations.md:18]: Alphabetize

**Review by keharper — CHANGES_REQUESTED** [review]
I don't know if you're doing this already, but be sure to run `rake` in the devdocs root directory to make sure that all links resolve.

**Comment by RakeshJesadiya** [comment]
@keharper I have updated the changes as per your suggestion. thanks.

File: src/guides/v2.3/graphql/interfaces/downloadable-product.md (hunk 1 above)
Comment (keharper) [reply src/guides/v2.3/graphql/interfaces/downloadable-product.md:22]: You missed this one. Also, you introduced an error in line 18 by linking to the wrong location.

File: src/guides/v2.4/graphql/interfaces/downloadable-product.md (hunk 5 above)
Comment (keharper) [reply src/guides/v2.4/graphql/interfaces/downloadable-product.md:21]: Same problem as the 2.3 version of the file

**Review by keharper — CHANGES_REQUESTED** [review]

**Comment by RakeshJesadiya** [comment]
@keharper I have updated the PR based on your suggestion. Thanks.

**Review by keharper — APPROVED** [review]

**Comment by keharper** [comment]
running tests

**Pull request merged**